### PR TITLE
Double factorial in C++ language

### PR DESCRIPTION
**Pull request by ag278**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

double factorial  of a number n (denoted by n!!) is the product of all the integers from 1 up to n that have the same parity(odd or even) as n.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#930

### CODE REVIEW

**Comment by ag278** [comment]
@iamrajiv  @Kajol-Kumari @siddharth25pandey @plazzy99  please review this
@iamrajiv I have added both the approaches as you mentioned in the issue.

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -68,11 +68,15 @@ _add list here_
 ## Mathematics
 
 - [Ackermann_Function](math/Ackermann_function.cpp)
+
```
Comment (atarax665) [inline C-Plus-Plus/README.md]:
```suggestion
```

File: C-Plus-Plus/README.md [hunk 2]
```diff
@@ -68,11 +68,15 @@ _add list here_
...
+- [Double factorial](math/double_factorial.cpp)
+
```
Comment (atarax665) [inline C-Plus-Plus/README.md]:
```suggestion
```

**Review by iamrajiv — APPROVED** [review]

**Comment by ag278** [comment]
@atarax665 I have made the required changes. Please review it

File: C-Plus-Plus/math/double_factorial.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    int ans = 1;
+    for (int i = num; i >= 0; i = i - 2) 
+    {
+        if (i == 0 || i == 1) 
+        {
+            return ans;
+        } 
+        else 
+        {
+            ans*=i;
```
Comment (atarax665) [inline C-Plus-Plus/math/double_factorial.cpp]:
```suggestion
            ans *= i;
```

File: C-Plus-Plus/math/double_factorial.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+    return ans;
+
```
Comment (atarax665) [inline C-Plus-Plus/math/double_factorial.cpp]:
```suggestion
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by ag278** [comment]
@atarax665 I have made the required changes.Please verify it

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Comment by ag278** [comment]
@plazzy99 could you please review my PR

**Review by vatsalkesarwani12 — APPROVED** [review]

**Pull request merged**